src/interface/state_descriptor.hpp

```diff
@@ -135,6 +177,8 @@ class StateDescriptor {
   const std::string label_;
   std::map<std::string, Metadata> metadataMap_;
   std::map<std::string, std::vector<Metadata>> sparseMetadataMap_;
+  std::map<std::string, Metadata> swarmMetadataMap_;
+  std::map<std::string, std::map<std::string, Metadata>> swarmValueMetadataMap_;
```

Comment:
Again if you don't need ordering unordered_map will probably be a better choice here.

Reply:
same thing above, leaving this for symmetry